docs: blog post, migration guide and roadmap (#228)
siemens/ix-docs: docs/home/migration/5_0_0/index.md

---
sidebar_position: 0
sidebar_title: Upgrade to V5
title: Upgrade to V5
doc-type: banner
description: Welcome to the migration guide for upgrading from Siemens Industrial Experience design system V4 to V5. This guide supports you through all the major changes.
---

# Upgrade to V5.0.0

:::info Recommended upgrade path
We strongly recommend upgrading version by version and not skipping major versions. If your project is still on an older major release, complete each intermediate migration first before moving to Version 5.
:::

> ## Migrate with your agent

Reviewer: since this is only applicable to code and you actually have a dedicated code section, why not put it under there?

This section gives an overview of the main code-level breaking changes in Version 5. For the complete list of changes, see the [Changelog](https://github.com/siemens/ix/releases) and the [Release V5.0.0 blog post](/blog/2026/05/11/release-5) for more context on the rationale behind the update.

### Migrate with your agent

> We optimized our [Breaking changes guide](https://github.com/siemens/ix/blob/main/BREAKING_CHANGES/v5.md) with a clear structure that is easy to parse both for humans and for agents.

Reviewer 2: ```suggestion
We optimized our [Breaking changes guide](https://github.com/siemens/ix/blob/main/BREAKING_CHANGES/v5.md) with a clear structure that is easy to parse both for humans and AI agents.
```

It starts with a high-level title and then splits the migration into isolated change domains, e.g. button accessibility, theming, [ix-application](/docs/components/application), [ix-toast](/docs/components/toast), and [ix-modals](/docs/components/modal). Within each domain, the guide follows a repeatable pattern:

- A short explanation of what changed and why

> - Exact API or attribute names that were removed, added, or renamed

Reviewer 2: ```suggestion
- Exact API or attribute names that were removed, added or renamed
```

- `ix-toast-container` no longer creates a separate wrapper element in `document.body`. The host element is now the container.

### Component updates

Several existing components now follow clearer and more consistent API patterns in Version 5.

- `ix-time-picker` renames the `i18n-column-header` attribute to `i18n-hour-column-header` to align with the rest of the time-related API.
- `ThemeSwitcher` now uses the new theme and color schema model and updates its event payloads accordingly.
- `ix-application` now separates the theme name from the color schema, replacing combined values, e.g. `classic-dark`.

### Global style updates

Theming is the largest migration topic in Version 5.

- We replaced legacy classes, e.g. `theme-classic-light` and `theme-classic-dark`, with `data-ix-theme` and `data-ix-color-schema` on the `html` element.
- iX no longer injects a default theme automatically, so applications should set both attributes explicitly if a default theme is required.
- The remaining Bootstrap font variable has been removed. If your application still relies on it, map `--bs-font-sans-serif` to `var(--theme-font-family)`.

## What changed in the Figma library

> Version 5 also introduces a broader set of updates in the iX Figma library. These changes improve visual consistency, introduce more flexible composition patterns, and align several properties and behaviors more closely with code. Make sure to update to the latest published version. You can review all design-related changes in detail in the Figma library’s release notes.

Reviewer 2: ```suggestion
Version 5 also introduces a broader set of updates in the iX Figma library. These changes improve visual consistency, introduce more flexible composition patterns, and align several properties and behaviors more closely with code. Review all design-related changes in detail in the Figma library’s release notes and ensure you update to the latest published version.  
```

### Design and visual updates

> - Siemens Reference Colors are now adopted across the library, refining status, interaction, and danger color styling.

Reviewer 2: ```suggestion
- Siemens Reference Colors are now adopted across the library, refining status, interaction, and danger color styling.
- 

big "C" for colors - proper noun here? If so, that's fine. 
```

Author: No, I changed to "Siemens reference colors", would not treat it as proper noun.

- Default, hover and active states are visually more distinct across multiple components, including buttons, chips, icon buttons, and toggle buttons.

> - Status-driven visuals are applied more consistently, e.g. through strokes or icons in components, e.g. blinds, cards, pills, and chips.

Reviewer 2: ```suggestion
- Status-driven visuals are applied more consistently, e.g. through strokes or icons in components, such as blinds, cards, pills, and chips.
```

### Structural and behavior changes

- Native Figma slots and slot-based layouts are introduced in key components such as [card](/docs/components/card) and [tabs](/docs/components/tabs), allowing more flexible and responsive compositions.

> - Internal building blocks such as [tab item](/docs/components/tabs) and other internal atoms are moved to internal-only components to simplify the public library surface.

Reviewer 2: ```suggestion
- Internal building blocks such as [tab item](/docs/components/tabs) and other internal atoms have been moved to internal-only components to simplify the public library surface.
```

> - Interaction behavior for scrolling, overflow, and selection has been updated in areas such as [tabs](/docs/components/tabs), [date dropdown](/docs/components/date-dropdown), and [chip](/docs/components/chip) close behavior.

Reviewer 2: ```suggestion
- Interaction behavior for scrolling, overflow and selection has been updated in areas such as [tabs](/docs/components/tabs), [date dropdown](/docs/components/date-dropdown), and [chip](/docs/components/chip) close behavior.
```

### Accessibility and usability improvements

- Critical status colors such as error and info are refined to better meet WCAG contrast requirements.
- Focus, hover, and active state handling is improved across form controls and other interactive elements.

> - Date-related inputs such as [date](/docs/components/input-date), [date time](/docs/components/input-date-time), and [date dropdown](/docs/components/date-dropdown) now provide clearer picker interactions and better separation of concerns.

Reviewer 2: ```suggestion
- Date-related inputs such as [date](/docs/components/input-date), [date time](/docs/components/input-date-time), and [date dropdown](/docs/components/date-dropdown) now provide clearer picker interactions and a better separation of concerns.
- 

*not sure what concerns mean here?? 
```

### Breaking changes and migration notes

> - [Card](/docs/components/card), action card, and push card are deprecated in favor of new card components that support slot-based layouts.

Reviewer: ```suggestion
- [Card](/docs/components/card), action card and push card are deprecated in favor of new card components that support slot-based layouts.
```
